primitives/src/vid/advz.rs

```diff
@@ -323,30 +350,36 @@ where
         let aggregate_proofs = UnivariateKzgPCS::multi_open_rou_proofs(
             &self.ck,
             &aggregate_poly,
-            self.num_storage_nodes,
+            code_word_size,
             &self.multi_open_domain,
         )
         .map_err(vid)?;
         end_timer!(agg_proofs_timer);
 
         let assemblage_timer = start_timer!(|| "assemble shares for dispersal");
-        let shares = all_storage_node_evals
-            .into_iter()
-            .zip(aggregate_proofs)
-            .enumerate()
-            .map(|(index, (evals, aggregate_proof))| {
-                Ok(Share {
+
+        let mut shares = Vec::with_capacity(code_word_size);
+        let mut evals = Vec::new();
+        let mut proofs = Vec::new();
+        for index in 0..code_word_size {
+            evals.extend(all_storage_node_evals[index].iter());
+            proofs.push(aggregate_proofs[index].clone());
+            if (index + 1) % self.multiplicity == 0 {
+                shares.push(Share {
                     index,
-                    evals,
-                    aggregate_proof,
-                    evals_proof: all_evals_commit
+                    evals: evals.clone(),
+                    aggregate_proofs: proofs.clone(),
```

Comment:
This might be a good use case for [`Vec::drain`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain). If that works then you could delete lines 378-379.

Reply:
Clippy is suggesting `std::mem::take(&mut evals)`.
But I'm unsure if introducing this has implications for our WASM compliance?
(Arkworks has a similar `ark_std::mem`)

Reply:
No idea. Tempting as it is to learn the answer, let's not get nerd-sniped by this question. I leave it to your judgement. 😁